src/main/webapp/app/shared/vertical-progress-bar/vertical-progress-bar.component.ts

```diff
@@ -0,0 +1,108 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, HostBinding, Input, OnInit } from '@angular/core';
+
+/**
+ * Simple Vertical Progress Bar without any external dependencies
+ * <ul>
+ *     <li> Supports three different colors depending on fill level
+ *     <li> [upperBorder, 100] -> upper color
+ *     <li> (lowerBorder, upperBorder) -> intermediate color
+ *     <li> [0,lowerBorder] -> lower color
+ * </ul>
+ */
+@Component({
+    selector: 'jhi-vertical-progress-bar',
+    templateUrl: './vertical-progress-bar.component.html',
+    styleUrls: ['./vertical-progress-bar.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VerticalProgressBarComponent implements OnInit {
```

Comment:
Can you elaborate why you chose a custom vertical implementation over the default Bootstrap progress bars?
https://ng-bootstrap.github.io/#/components/progressbar/examples

Reply:
To add to that, the Rating component (e.g., with a student icon) could also work great here. 
https://ng-bootstrap.github.io/#/components/rating/examples#decimal

Reply:
I just wanted to play around a little bit with css. See no harm in it

Reply:
What do the other reviewers say? @simon-hng @Strohgelaender